### PR TITLE
Add Support for Authentication via Secure Token/Config File

### PR DESCRIPTION
**Pull request by sgiehl**
### Summary

This PR improves authentication security for import_logs.py and updates documentation to match actual CLI behavior.

The importer previously required credentials via CLI flags (--token-auth, --login, --password), which can expose secrets in process listings (ps, htop) on shared systems.
This PR adds a safer config-file based auth path and deprecates CLI secret input.

#### Auth code changes

- Added --auth-config=<path> support.
- Implemented auth config parsing from [auth]:
    - token_auth = ...
    - or login = ... + password = ...
- Added permission checks for auth config files and warning when permissions are too open (recommended 600).
- Added deprecation warnings when secrets are passed via CLI:
    - --token-auth
    - --login
    - --password
- Updated related error messages to recommend --auth-config.

#### Tests

Added auth-focused tests for:

- loading token from --auth-config
- loading login/password from --auth-config and fetching token via API call
- deprecation warning emission for insecure CLI secret flags
- warning for overly permissive auth config permissions

#### README updates

- Corrected docs to match actual CLI/auth behavior.
- Added/updated secure auth guidance around --auth-config.
- Corrected outdated examples/flags and usage notes.
- Clarified authentication method behavior and fixed stale references.

### Backward Compatibility

- Existing CLI auth flags still work for now, but are now explicitly deprecated and warned.
- No runtime behavior removed in this PR.

### Checklist

- [✔] I have understood, reviewed, and tested all AI outputs before use
- [✔] All AI instructions respect security, IP, and privacy rules

### Review

- [ ] [Functional review done](https://developer.matomo.org/guides/pull-request-reviews#functional-review-done)
- [ ] [Potential edge cases thought about](https://developer.matomo.org/guides/pull-request-reviews#potential-edge-cases-thought-about) (behaviour of the code with strange input, with strange internal state or possible interactions with other Matomo subsystems)
- [ ] [Usability review done](https://developer.matomo.org/guides/pull-request-reviews#usability-review-done) (is anything maybe unclear or think about anything that would cause people to reach out to support)
- [ ] [Security review done](https://developer.matomo.org/guides/security-in-piwik#checklist)
- [ ] [Wording review done](https://developer.matomo.org/guides/pull-request-reviews#translations-wording-review-done)
- [ ] [Code review done](https://developer.matomo.org/guides/pull-request-reviews#code-review-done)
- [ ] [Tests were added if useful/possible](https://developer.matomo.org/guides/pull-request-reviews#tests-were-added-if-usefulpossible)
- [ ] [Reviewed for breaking changes](https://developer.matomo.org/guides/pull-request-reviews#reviewed-for-breaking-changes)
- [ ] [Developer changelog updated if needed](https://developer.matomo.org/guides/pull-request-reviews#developer-changelog-updated-if-needed)
- [ ] [New documentation added or existing documentation updated](https://developer.matomo.org/guides/pull-request-reviews#documentation-added-if-needed)

### CODE REVIEW

File: import_logs.py [hunk 1]
```diff
@@ -1078,6 +1087,120 @@ def _valid_date(self, value):
...
+        )
+
+    def _apply_auth_precedence(self, argv):
+        cli_has_token_auth = self._has_option(argv, '--token-auth')
+        cli_has_login = self._has_option(argv, '--login')
+        cli_has_password = self._has_option(argv, '--password')
+
+        auth_config_credentials = None
+        if self.options.auth_config_file:
+            auth_config_credentials = self._load_auth_config(self.options.auth_config_file)
```
Comment (chippison) [inline import_logs.py]: Loading the `auth-config` file here can break precedence rule (token-auth, login+password, auth-config, config) when the auth-config file is either invalid or missing and either token-auth, or login+password are passed in. 

Example command that results in a `Fatal Error`:
```
python3 import_logs.py --url=https://matomo.ddev.site --idsite=1 --auth-config=/tmp/my-missing.cfg  --login=mylogin --password=mypassword access.log
```
Would also be good to add a test for this edge case for future checks.

**Review by chippison — COMMENTED** [review]
Reviewed and done some Functional Tests and is all good.

There is just an edge case issue/bug found above

**Comment by sgiehl** [comment]
@chippison I've applied some fixes.

**Review by chippison — APPROVED** [review]
